integration/e2e/service.go

```diff
@@ -334,21 +350,32 @@ func NewCommandWithoutEntrypoint(cmd string, args ...string) *Command {
 	}
 }
 
-type ReadinessProbe struct {
+type ReadinessProbe interface {
+	Ready(service *ConcreteService) (err error)
+}
+
+// HTTPReadinessProbe checks readiness by making HTTP call and checking for expected HTTP status code
+type HTTPReadinessProbe struct {
 	port           int
 	path           string
 	expectedStatus int
 }
 
-func NewReadinessProbe(port int, path string, expectedStatus int) *ReadinessProbe {
-	return &ReadinessProbe{
+func NewHTTPReadinessProbe(port int, path string, expectedStatus int) *HTTPReadinessProbe {
+	return &HTTPReadinessProbe{
 		port:           port,
 		path:           path,
 		expectedStatus: expectedStatus,
 	}
 }
 
-func (p *ReadinessProbe) Ready(localPort int) (err error) {
+func (p *HTTPReadinessProbe) Ready(service *ConcreteService) (err error) {
+	// Map the container port to the local port
+	localPort, ok := service.networkPortsContainerToLocal[p.port]
```

Comment:
This should be readiness probe's responsability. What do you think to add a function `ConcreteService.EndpointPort()` which is like `.Endpoint()` but just returns the port? We could return `0` if the port doesn't exist and convert the following `if` from `if !ok` to `if port == 0`.

What's your take?

Reply:
Errata corrige. You need the endpoint here, so you can directly use `Endpoint()`.